### ISSUE TITLE
Extend smart command parsing to not interpret line as command when it contains =

### ISSUE THREAD

**Issue by antocuni** [pdbpp/pdbpp#27]
Originally reported by: **Marc Abramowitz (Bitbucket: [msabramo](https://bitbucket.org/msabramo), GitHub: [msabramo](https://github.com/msabramo))**

----------------------------------------

I love the smart command parsing so that if I have a variable named `r` and I hit `r` to print it, it doesn't run the program.

However, another case where this doesn't work is if I want to assign a variable that doesn't yet exist:

e.g.:

```python
> /Users/marca/dev/git-repos/pip/pip/req.py(1296)install()
-> to_install = [r for r in self.requirements.values()
(Pdb++) self.requirements
Requirements({'smlib.log': <pip.req.InstallRequirement object at 0x101a50c50>})
(Pdb++) r = self.requirements['smlib.log']
Installing collected packages: smlib.log
  Running setup.py install for smlib.log
```

Note that I try to assign to `r` here so that I can do more inspection but because `r` doesn't exist, smart command parsing doesn't kick in and it runs the program.

It would be great if it could detect the `=` in the line and not run.

----------------------------------------
- Bitbucket: https://bitbucket.org/antocuni/pdb/issue/27

**Comment by antocuni** [comment]
*Original comment by* **Marc Abramowitz (Bitbucket: [msabramo](https://bitbucket.org/msabramo), GitHub: [msabramo](https://github.com/msabramo))**:

----------------------------------------

Oh, hmmmm, is assigning not supported?

```python
(Pdb++) req = self.requirements['smlib.log']
(Pdb++) req
*** NameError: name 'req' is not defined
```

**Comment by antocuni** [comment]
*Original comment by* **Marc Abramowitz (Bitbucket: [msabramo](https://bitbucket.org/msabramo), GitHub: [msabramo](https://github.com/msabramo))**:

----------------------------------------

Ah, the issue in my above comment with the assignment not taking effect seems to be a bug in Python 2.6 -- see:

* http://stackoverflow.com/questions/7912820/how-do-i-change-a-value-while-debugging-python-with-pdb
* http://bugs.python.org/issue5215

This works fine in Python 2.7.

**Comment by antocuni** [comment]
*Original comment by* **Marc Abramowitz (Bitbucket: [msabramo](https://bitbucket.org/msabramo), GitHub: [msabramo](https://github.com/msabramo))**:

----------------------------------------

I have a solution in this PR:

https://bitbucket.org/antocuni/pdb/pull-request/12/make-pep-8-compliant-improve-python-3/diff

**Comment by antocuni** [comment]
*Original comment by* **Antonio Cuni (Bitbucket: [antocuni](https://bitbucket.org/antocuni), GitHub: [antocuni](https://github.com/antocuni))**:

----------------------------------------

Don't treat lines like "r = 5" or "c = 6" as commands

Fixes issue #27 ("Extend smart command parsing to not interpret line as command
when it contains =")
